service/src/main/java/bio/terra/tanagra/app/configuration/ExportConfiguration.java

```diff
@@ -0,0 +1,126 @@
+package bio.terra.tanagra.app.configuration;
+
+import bio.terra.tanagra.service.export.DataExport;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@EnableConfigurationProperties
+@ConfigurationProperties(prefix = "tanagra.export")
+public class ExportConfiguration {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ExportConfiguration.class);
+
+  private Shared shared;
+  private List<PerModel> models;
+
+  public Shared getShared() {
+    return shared;
+  }
+
+  public void setShared(Shared shared) {
+    this.shared = shared;
+  }
+
+  public List<PerModel> getModels() {
+    return Collections.unmodifiableList(models);
+  }
+
+  public void setModels(List<PerModel> models) {
+    this.models = models;
+  }
+
+  /** Write the data export flags into the log. Add an entry here for each new flag. */
+  public void logConfig() {
+    LOGGER.info("Export shared gcs-project-id: {}", shared.getGcsProjectId());
+    LOGGER.info(
+        "Export shared gcs-bucket-names: {}",
+        shared.getGcsBucketNames().stream().collect(Collectors.joining(",")));
+    for (int i = 0; i < models.size(); i++) {
+      PerModel m = models.get(i);
+      LOGGER.info("Export models[{}] name: {}", i, m.getName());
+      LOGGER.info("Export models[{}] display-name: {}", i, m.getDisplayName());
+      LOGGER.info("Export models[{}] type: {}", i, m.getType());
+      LOGGER.info("Export models[{}] redirect-away-url: {}", i, m.getRedirectAwayUrl());
+      LOGGER.info(
+          "Export models[{}] params: {}",
+          i,
+          m.getParams().stream().collect(Collectors.joining(",")));
+    }
+  }
+
+  public static class Shared {
+    private String gcsProjectId;
+
+    private List<String> gcsBucketNames;
+
+    public String getGcsProjectId() {
```

Comment:
Are these parameters not specific to individual plugins? They wouldn't be necessary for something like an AWS export, right?

Reply:
Right, they're just shared infrastructure that all plugins can use. Equivalent to duplicating these parameters for each export model. I was thinking of this shared config as useful for terraform outputs, and the per-model config as useful for things like URLs/filename prefixes/etc.